rllib/agents/trainer.py

```diff
@@ -1002,12 +1002,15 @@ def setup(self, config: PartialTrainerConfigDict):
 
             self.config["evaluation_config"] = eval_config
 
+            env_id = self._register_if_needed(eval_config.get("env"), eval_config)
```

Comment:
wait, what if eval_config doesn't have env key? and they just want to use the same env as rollout workers?
won't env_id and env_creator be None here?

Reply:
`eval_config` is a copy of the main config plus the eval_config overrides given by the user, so the key is definitely there at this point in the code.
If `env` is None, it's also not a problem, then the `env_id` will be None and the env creator will return None.

Reply:
ah I see. thanks for the explanation.